syntax = "proto3";

package user;

service UserService {
  rpc SendUser (User) returns (Response);
    rpc StreamUsers(stream User) returns (stream Response);
}

service FileService {
  rpc Upload(stream FileChunk) returns (UploadStatus);
}

message User {
  string _id = 1;
  int32 index = 2;
  string guid = 3;
  bool isActive = 4;
  string balance = 5;
  int32 age = 6;
  string eyeColor = 7;
  string name = 8;
  string gender = 9;
  string company = 10;
  string email = 11;
  string phone = 12;
  string address = 13;
  string about = 14;
  string registered = 15;
  double latitude = 16;
  double longitude = 17;
}

message FileChunk {
  bytes content = 1;
}

message UploadStatus {
  string message = 1;
}

message Response {
  string message = 1;
}
